alf/data_structures.py

```diff
@@ -110,6 +111,37 @@ def is_last(self):
         raise ValueError('step_type is not a Torch Tensor')
 
 
+def _create_timestep(observation, prev_action, reward, discount, env_id,
+                     step_type):
+    discount = torch.tensor(discount)
+    # as_tensor reuses the underlying data store of numpy array if possible.
+    create_tensor = lambda t: t.detach() if torch.is_tensor(
+        t) else torch.as_tensor(t).detach()
```

Comment:
detach() is not necessary for torch.as_tensor(t) because numpy doesn't support gradients

Reply:
Good point, done!